webapp/client/src/components/FormFieldIntegerInput.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { IMaskInput } from "react-imask";
+import classNames from "classnames";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+import FieldLabel from "./FieldLabel";
+import FormFieldSeparatedField from "./FormFieldSeparatedField";
+
+function FormFieldTextInput({
+  fieldName,
+  fieldId,
+  value,
+  required,
+  autofocus,
+  label,
+  maxWidth,
+  maxLength,
+  details,
+  errors,
+}) {
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+      }}
+      labelComponent={<FieldLabel {...{ label, fieldId, details }} />}
+      render={() => (
+        <IMaskInput
+          mask={Number}
+          scale={0}
+          type="text"
+          id={fieldId}
+          name={fieldName}
+          defaultValue={value}
+          inputMode="numeric"
+          pattern="[0-9]*"
+          maxLength={maxLength}
+          // TODO: autofocus is under review.
+          // eslint-disable-next-line
+          autoFocus={autofocus || Boolean(errors.length)}
+          required={required}
+          className={classNames("form-control", `input-width-${maxWidth}`)}
+        />
+      )}
+    />
+  );
+}
+
+FormFieldTextInput.propTypes = {
+  fieldId: PropTypes.string.isRequired,
+  fieldName: PropTypes.string.isRequired,
+  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
+  autofocus: PropTypes.bool,
+  required: PropTypes.bool,
+  setMask: PropTypes.bool,
+  value: PropTypes.string.isRequired,
+  label: FieldLabel.propTypes.label,
+  maxWidth: PropTypes.number,
+  maxLength: PropTypes.number,
```

Comment:
> Can you explain why maxWidth is needed now? In all previous cases I encountered, this was derived from the maxLength.

That's not always the case. I specifically remember Nadine requesting that the component is optically limited, but you should be able to input more that the requested number of chars. (mostly because of accessibility reasons)

> Also is this really setting a maximum for the width (as in: can it ever be less)?
Very good question. In fact, it is the width itself. So I'll rename it to `fieldWidth` then.